clang/test/SemaCXX/overloaded-operator.cpp

```diff
@@ -598,3 +598,30 @@ namespace B {
 }
 void g(B::X x) { A::f(x); }
 }
+
+namespace GH78314 {
+
+class a {
+public:
+  void operator--() = delete; // expected-note {{candidate function has been explicitly deleted}} \
+                              // expected-note {{candidate function not viable: requires 0 arguments, but 1 was provided}}
+  void operator--(int) = delete; // expected-note {{candidate function has been explicitly deleted}} \
+                                 // expected-note {{candidate function not viable: requires 1 argument, but 0 were provided}}
+};
```

Comment:
can you add tests for (in another class)

```cpp
void operator--(this a) = delete;
void operator--(this a, int) = delete
``` 
?

Thanks!

Reply:
Sure, done.